### ISSUE TITLE
Segmentation Fault When Generating Document

### ISSUE THREAD

**Issue by cheapcontact** [wkhtmltopdf/wkhtmltopdf#3966]
Here is the command that causes it:
/u04/wkhtmltox/bin/wkhtmltopdf --disable-smart-shrinking -L 13mm -R 13mm -T 15.4mm  -B 18.5mm  --footer-html  https://dev.surveyorhealth.com/dcls/PCP/ReportFooter  https://dev.surveyorhealth.com/debug/busted_report_html.htm PCPPCP680031529360688428.pdf

Here is my system info:
  Operating System: Oracle Linux Server 7.4
       CPE OS Name: cpe:/o:oracle:linux:7:4:server
            Kernel: Linux 4.1.12-61.1.33.el7uek.x86_64
      Architecture: x86-64

Here is the wkhtmltopdf version info:
wkhtmltopdf 0.12.4 (with patched qt)

Some information about the error.
If I remove the footer or the margins or the ignore smart shrink option, then the error goes away.
There is a table in the html document that spans an entire page labeled "Current Medications." If I remove a row from that table the error goes away.

**Comment by ashkulz** [comment]
Duplicate of #2711 -- update to 0.12.5

**Comment by atekie** [comment]
I know this is old, however, it is still happening for certain use cases.

I am running an AWS beanstalk application (PHP 7.2 running on 64bit Amazon Linux/2.8.9) with 0.12.5 installed.  I am using wkhtmltopdf to print confidential information so unfortunately, I will not be able to share the actual content I am printing.

/usr/local/bin/wkhtmltopdf --header-html 'http://xxx.xxx.xxx.xxx' --footer-html 'http://xxx.xxx.xxx.xxx' --no-outline --margin-top '50' --margin-right '5' --margin-bottom '15' --margin-left '5' --disable-smart-shrinking cover 'http://xxx.xxx.xxx.xxx' 'http://bd-lms.com/xxx.xxx.xxx.xxx' '/tmp/tmp_wkhtmlto_pdf_cjFfkQ.pdf'

I receive the following error (only for certain documents):

Loading pages (1/6)
Counting pages (2/6)
Resolving links (4/6)
Loading headers and footers (5/6)
Printing pages (6/6)
Segmentation fault===========================================] Page 6 of 5

If I remove the footer margin setting or remove 'disable-smart-shrinking' the PDF prints fine.

Most documents print fine, so it appears to be content specific.

**Comment by atekie** [comment]
I know this is old, however, it is still happening for certain use cases.

I am running an AWS beanstalk application (PHP 7.2 running on 64bit Amazon Linux/2.8.9) with 0.12.5 installed. I am using wkhtmltopdf to print confidential information so unfortunately, I will not be able to share the actual content I am printing.

/usr/local/bin/wkhtmltopdf --header-html 'http://xxx.xxx.xxx.xxx' --footer-html 'http://xxx.xxx.xxx.xxx' --no-outline --margin-top '50' --margin-right '5' --margin-bottom '15' --margin-left '5' --disable-smart-shrinking cover 'http://xxx.xxx.xxx.xxx' 'http://xxx.xxx.xxx.xxx' '/tmp/tmp_wkhtmlto_pdf_cjFfkQ.pdf'

I receive the following error (only for certain documents):

Loading pages (1/6)
Counting pages (2/6)
Resolving links (4/6)
Loading headers and footers (5/6)
Printing pages (6/6)
Segmentation fault===========================================] Page 6 of 5

If I remove the footer margin setting or remove 'disable-smart-shrinking' the PDF prints fine.

Most documents print fine, so it appears to be content specific.

**Comment by joshuapinter** [comment]
Not quite the same but posting here because this is where my rabbit hole took me and it may help others. 

We're generating fairly large PDFs and were running into a segmentation fault at ~1,000 pages.

After a thorough investigation, there were a few things that fixed the issue: 

1. Removing the image in the footer. 
2. Removing either the current page or the total pages from the footer.
3. Removing the unused variables, like `section` and `subsection`, from the `subst` function that substitutes the page counts, as specified in the [docs](https://wkhtmltopdf.org/usage/wkhtmltopdf.txt).

After doing **just** 3 (and leaving the image and current page and total pages in the footer), we were then able to generate PDFs past 7,000 pages without a problem. 

I don't know the inner workings enough to know the real culprit, but it seems in general that it was more taxing on the process(es) to parse and look for each of those variables from the GET param when you get to large PDFs.

In short, only parse the ones you're using.
